src/main/scala/analysis/Cfg.scala

```diff
@@ -426,7 +427,9 @@ class ProgramCfgFactory:
       cfg.addEdge(funcEntryNode, funcExitNode)
     } else {
       // Recurse through blocks
-      visitBlock(proc.blocks.head, funcEntryNode)
+      visitBlock(proc.entryBlock.get, funcEntryNode)
+      // If it has no entry-block we still visit the exit block because VSA analysis expects everything to have an Exit
+      visitBlock(proc.returnBlock, funcEntryNode)
```

Comment:
If the procedure has no entry block, an exception will be thrown by entryBlock.get, so this isn't really doing what you want at all anyway.